### PR TITLE
[ML] Adding more logging for issue #76075

### PR DESCRIPTION
**Pull request by benwtrent**
While investigating test failure #76075, it became evident that there is one completed bucket missing in the composite agg results. 

This commit adds logging to print out all the buckets (and their counts) whenever the finalized bucket count doesn't match up.

Additionally, instead of using chunking, this simply adjusts the composite aggregation paging size.

related: https://github.com/elastic/elasticsearch/issues/76075

### CODE REVIEW

File: x-pack/plugin/ml/qa/native-multi-node-tests/src/javaRestTest/java/org/elasticsearch/xpack/ml/integration/DatafeedJobsIT.java [hunk 1]
```diff
@@ -390,10 +389,21 @@ public void testStopAndRestartCompositeDatafeed() throws Exception {
...
+                    Collections.singletonList(
+                        new DateHistogramValuesSourceBuilder("timebucket")
+                            .fixedInterval(new DateHistogramInterval("1h"))
+                            .field("time")
+                    )
+                ).subAggregation(AggregationBuilders.max("time").field("time")).size(100)
+            );
             updateDatafeed(new DatafeedUpdate.Builder()
                 .setId(compositeDatafeedConfig.getId())
                 // Set to auto to speed up and finish the job
```
Comment (przemekwitek) [inline x-pack/plugin/ml/qa/native-multi-node-tests/src/javaRestTest/java/org/elasticsearch/xpack/ml/integration/DatafeedJobsIT.java:405]: This comment is not relevant anymore, is it?

File: x-pack/plugin/ml/qa/native-multi-node-tests/src/javaRestTest/java/org/elasticsearch/xpack/ml/integration/DatafeedJobsIT.java [hunk 2]
```diff
@@ -405,10 +415,16 @@ public void testStopAndRestartCompositeDatafeed() throws Exception {
 
         List<Bucket> scrollBuckets = getBuckets(scrollJobId);
         List<Bucket> compositeBuckets = getBuckets(compositeJobId);
+        assertThat(
+            "composite bucket size " + scrollBuckets + " does not equal scroll bucket size" + compositeBuckets,
```
Comment (przemekwitek) [inline x-pack/plugin/ml/qa/native-multi-node-tests/src/javaRestTest/java/org/elasticsearch/xpack/ml/integration/DatafeedJobsIT.java:419]: The order is wrong I think. `scrollBuckets` and `compositeBuckets` should be swapped.

File: x-pack/plugin/ml/qa/native-multi-node-tests/src/javaRestTest/java/org/elasticsearch/xpack/ml/integration/DatafeedJobsIT.java (hunk 1 above)
Comment (benwtrent) [inline x-pack/plugin/ml/qa/native-multi-node-tests/src/javaRestTest/java/org/elasticsearch/xpack/ml/integration/DatafeedJobsIT.java:405]:
```suggestion
```

File: x-pack/plugin/ml/qa/native-multi-node-tests/src/javaRestTest/java/org/elasticsearch/xpack/ml/integration/DatafeedJobsIT.java (hunk 2 above)
Comment (benwtrent) [inline x-pack/plugin/ml/qa/native-multi-node-tests/src/javaRestTest/java/org/elasticsearch/xpack/ml/integration/DatafeedJobsIT.java:419]:
```suggestion
            "scroll bucket size " + scrollBuckets + " does not equal composite bucket size" + compositeBuckets,
```

**Review by przemekwitek — APPROVED** [review]
LGTM

**Pull request merged**